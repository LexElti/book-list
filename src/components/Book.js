import React from 'react'

export default ({book}) => {
  return (
    <div className="col-lg-3 col-md-4 col-sm-6 mb-4">
      <div className="card">
        {book.thumbnail  
          ? <img src={book.thumbnail} className="card-img-top" />
          : <div className="card-img-top"></div>
        }
        <div className="card-body">
          <h5 className="card-title">{book.title}</h5>
          <p className="card-text">Authors: {book.authors.join(', ')}</p>
          <p className="card-text">{book.description}</p>
          <a 
            href={book.previewLink}
            className="btn btn-primary"
            target="_blank"
            rel="noopener noreferrer"
          >Preview</a>
        </div>
      </div>
    </div>
  )
}
