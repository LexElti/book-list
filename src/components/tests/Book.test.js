import React from 'react'
import Book from '../Book'

describe('<Book />', () => {
  const props = {
    book: {
      title: 'Title',
      authors: [
        'Author1', 'Author2'
      ],
      description: 'Description',
      previewLink: 'previewLink',
      thumbnail: 'thumbnail'
    }
  }

  describe('<Book /> main test with thumbnail', () => {
    const wrapper = shallow(<Book {...props} />)

    it('renders properly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('correct render data', () => {
      expect(wrapper.find('h5').text()).toEqual(props.book.title)
      expect(
        wrapper.find('.card-body').find('p').first().text()
      ).toEqual(`Authors: ${props.book.authors.join(', ')}`)
      expect(
        wrapper.find('.card-body').find('p').at(1).text()
      ).toEqual(props.book.description)
      expect(
        wrapper.find('a').prop('href')
      ).toEqual(props.book.previewLink)
      expect(
        wrapper.find('img').prop('src')
      ).toEqual(props.book.thumbnail)
    })
  })

  describe('<Book /> without thumbnail', () => {
    const nextProps = {
      ...props,
      book: {
        ...props.book,
        thumbnail: ''
      }
    }

    const wrapper = shallow(<Book {...nextProps} />)

    it('renders properly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('correct render data', () => {
      expect(wrapper.find('div.card-img-top')).toBeDefined()
    })
  })
})
