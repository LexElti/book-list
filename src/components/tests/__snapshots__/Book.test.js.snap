// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> <Book /> main test with thumbnail renders properly 1`] = `ShallowWrapper {}`;

exports[`<Book /> <Book /> without thumbnail renders properly 1`] = `ShallowWrapper {}`;
